kernel/src/fs/fs_resolver.rs

```diff
@@ -176,6 +178,13 @@ impl FsResolver {
 
         Ok(lookup_res)
     }
+        fn check_dir_search_permission(path: &Path) -> Result<()> {
+        let inode = path.inode();
+        if inode.type_() == InodeType::Dir {
+            inode.check_permission(Permission::MAY_EXEC)?;
```

Comment:
This check is already included in the `Path::lookup` method:

https://github.com/asterinas/asterinas/blob/70eda539dfade526c28d44e6cb60c056c56cc06c/kernel/src/fs/path/mod.rs#L73-L82